From da6805b1a1cf30fe48b350858ebd9d82ee1adf45 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 27 Jan 2025 04:55:49 +0800
Subject: [PATCH 2/5] AOSCOS: LOONGSON3: 01systemd-crypsetup: remove TPM2
 dependency

There is just no way that we will get TPM2 on MIPS-based Loongson 3
systems - and it saves almost 10MiB post-XZ.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 modules.d/01systemd-cryptsetup/module-setup.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/modules.d/01systemd-cryptsetup/module-setup.sh b/modules.d/01systemd-cryptsetup/module-setup.sh
index 034d4ecf..00a59ff5 100755
--- a/modules.d/01systemd-cryptsetup/module-setup.sh
+++ b/modules.d/01systemd-cryptsetup/module-setup.sh
@@ -31,7 +31,13 @@ depends() {
             deps+=" tpm2-tss"
         fi
     elif [[ ! $hostonly ]]; then
-        for module in fido2 pkcs11 tpm2-tss; do
+        if [[ ${DRACUT_ARCH:-$(uname -m)} != mips64 ]]; then
+            modules="fido pkcs11 tpm2-tss"
+        else
+            # MIPS64: No chance for TPM2 on these platforms.
+            modules="fido pkcs11 tpm2-tss"
+        fi
+        for module in "$modules"; do
             module_check $module > /dev/null 2>&1
             if [[ $? == 255 ]]; then
                 deps+=" $module"
-- 
2.48.1

